core/src/main/java/fr/sncf/osrd/api/pathfinding/PathfindingResultConverter.java

```diff
@@ -96,8 +105,12 @@ public static List<PathWaypointResult> getWaypointsOnRoute(
                     var waypoints = new ArrayList<PathWaypointResult>();
 
                     // Add operational points as optional waypoints
-                    for (var op : truncatedRange.getOperationalPoints())
-                        waypoints.add(PathWaypointResult.suggestion(op.element(), trackSection, op.offset()));
+                    for (var op : truncatedRange.getOperationalPoints()) {
+                        // The offset of the route + The offset of the track + the offset of the op
+                        var pathOffset = routeOffset + offset + op.offset();
```

Comment:
I think this only works when the path goes in the same direction as the track section